cmake/deps-helper.cmake

```diff
@@ -0,0 +1,46 @@
+# where to build dependencies
+set(DEPS_PREFIX "${CMAKE_BINARY_DIR}/deps")
+# base path to installed dependencies
+set(DEPS_BASE "${CMAKE_BINARY_DIR}/deps")
```

Comment:
Shouldn't this be something like `${CMAKE_BINARY_DIR}/deps/install` otherwise it is the same as `DEPS_PREFIX`?

Reply:
You can do either, I guess. Note that all checkouts and build folders etc are in `${CMAKE_BINARY_DIR}/deps/src/`, thus I felt it was okay to avoid this additional folder... but I don't really care.

Reply:
It probably just matters for caching in CI. Do we have to cache the `${CMAKE_BINARY_DIR}/deps/` or only the `${CMAKE_BINARY_DIR}/deps/{lib,include,..}` directories?

Reply:
Right now we need to cache `deps/src` as well, i.e. `${CMAKE_BINARY_DIR}/deps/`.
It's even worse than "not being cached": removing `deps/src` from a succesful build breaks the subsequent build for weird reasons...

It may be possible to change this (maybe via `BUILD_BYPRODUCTS`?) but it seems rather unlikely to me. Note that the `src` directly also contains the `stamp` files that store the timestamps necessary to determine whether something needs to be updated...
I'd suggest to simply cache the whole directory (just like we cached the whole `deps` directory beforehand)

Reply:
I just hope that the whole directory won't be too large (it includes all the object files, sources etc.). We just tried to get down the size of the caches so that we can store more ccache caches.

Reply:
As per offline discussion: the folder size is of course larger than only the install folder of the previous deps folder, but still okay.